airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultCheckConnectionWorker.java

```diff
@@ -73,18 +70,12 @@ public ConnectorJobOutput run(final StandardCheckConnectionInput input, final Pa
           WorkerConstants.SOURCE_CONFIG_JSON_FILENAME,
           Jsons.serialize(input.getConnectionConfiguration()));
 
-      LineGobbler.gobble(process.getErrorStream(), LOGGER::error);
-
-      final Map<Type, List<AirbyteMessage>> messagesByType;
-      try (final InputStream stdout = process.getInputStream()) {
-        messagesByType = streamFactory.create(IOs.newBufferedReader(stdout))
-            .collect(Collectors.groupingBy(AirbyteMessage::getType));
+      final ConnectorJobOutput jobOutput = new ConnectorJobOutput().withOutputType(OutputType.CHECK_CONNECTION);
 
-        WorkerUtils.gentleClose(process, 1, TimeUnit.MINUTES);
-      }
+      LineGobbler.gobble(process.getErrorStream(), LOGGER::error);
 
-      final int exitCode = process.exitValue();
-      final Optional<AirbyteConnectionStatus> status = messagesByType
+      final Map<Type, List<AirbyteMessage>> messagesByType = WorkerUtils.getMessagesByType(process, streamFactory, 30);
+      final Optional<AirbyteConnectionStatus> connectionStatus = messagesByType
           .getOrDefault(Type.CONNECTION_STATUS, new ArrayList<>()).stream()
           .map(AirbyteMessage::getConnectionStatus)
           .findFirst();
```

Comment:
Want to make sure my understanding is correct - here we look at all of the messages getting sent though, filter for connection status messages and grab the connection status from those, but only use the first one as the `connectionStatus`. Do we expect to have multiple of these messages and if so, why does the first one accurately represent the connection status?

Reply:
I think the `first` makes sense because the connector should output a single status message. But this is not something the platform should be aware of. I think picking the last status instead of the first one would be safer. I will create an issue.

Reply:
https://github.com/airbytehq/airbyte/issues/21400

Reply:
Yeah, that seems like a very implicit expectation! Agree that the last probably is more likely to represent the actual status (e.g. if we for some reason output status messages for multiple retries while trying to connect). Thanks for making an issue